### ISSUE TITLE
Avoid usage of org.json library

### ISSUE THREAD

**Issue by chetanmeh** [Azure/azure-cosmosdb-java#29]
The driver currently has a dependency on `org.json`

```xml
 <dependency>
      <groupId>org.json</groupId>
      <artifactId>json</artifactId>
      <version>20140107</version>
</dependency>
```

This library license is marked as [Category-X][1] and hence such a library [cannot be used as a dependency in Apache Project][2]. Would it be possible to replace this library with some other json library? Otherwise it would become tricky to use azure-cosmosdb-java for Apache projects

[1]: https://www.apache.org/legal/resolved.html#category-x
[2]: https://www.apache.org/legal/resolved.html#optional

**Comment by moderakh** [comment]
@chetanmeh are you considering using the sdk as a dependency for a open source apache project? or is your question more of a general question?

**Comment by chetanmeh** [comment]
@moderakh  We plan to use this  driver for implementing a CosmosDB backend for [Apache OpenWhisk][1]. 

[1]: https://openwhisk.apache.org/

**Comment by moderakh** [comment]
@chetanmeh you opened also another issue about moving to RxJava2. Moving away from org.json can be done with less effort comparing to moving RxJava2.

I wonder if we replace org.json with some other apache friendly dependency, that will unblock you for now?

**Comment by chetanmeh** [comment]
> I wonder if we replace org.json with some other apache friendly dependency, that will unblock you for now?

+1. That would be very helpful as `org.json` dependency is a blocker for us. RxJava update is something that can be picked up later given that its already being worked upon

**Comment by moderakh** [comment]
@srinathnarayanan we should be able to rely only on jackson lib and drop the dependency on org.json lib.

I am assigning this to you. Please take a look.

**Comment by otaviojava** [comment]
Just my two cents: One suggestion would be json-p

JSON-P is a Java request specification, [JSR 353](https://jcp.org/en/jsr/detail?id=353) , and it has multiples implementations, that includes Jackson itself.  Also, the user is able to replace the implementation without impact the code if they want to.


* https://javaee.github.io/jsonp/
* GSON: https://code.google.com/archive/p/genson/wikis/JSR353.wiki
* https://github.com/FasterXML/jackson/wiki/JacksonRelease1.5

**Comment by chetanmeh** [comment]
@moderakh @srinathnarayanan Would it be possible to share some details on when a sdk release with json changes can be expected?

**Comment by moderakh** [comment]
@chetanmeh version sdk 2.0.0 with the changes for removing org.json is released to maven:
http://central.maven.org/maven2/com/microsoft/azure/azure-cosmosdb/2.0.0/

Thanks for your help and collaboration.

**Comment by chetanmeh** [comment]
Many thanks @moderakh  for accommodating such a major change! Now OpenWhisk CosmosDB PR can be merged :)
